src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -62,13 +64,52 @@ public static bool TryGetCounterPayload(this TraceEvent traceEvent, CounterFilte
                     displayUnits,
                     value,
                     counterType,
-                    intervalSec);
+                    intervalSec,
+                    metadataDict);
                 return true;
             }
 
             return false;
         }
 
+        public static Dictionary<string, string> GetMetadata(string metadataPayload)
```

Comment:
nit: internal, since it's not really an API but only visible to simplify unit testing.